build_runner_core/lib/src/changes/build_script_updates.dart

```diff
@@ -94,3 +106,16 @@ class BuildScriptUpdates {
     return null;
   }
 }
+
+/// Always returns false for [hasBeenUpdated], used when we want to skip
+/// the build script checks.
+class _NoopBuildScriptUpdates implements BuildScriptUpdates {
```

Comment:
[optional] Could avoid the `UnimplementedError()` required methods if we make both implementations different classes:

```dart
abstract class BuildScriptUpdates {
  static Future<BuildScriptUpdates> create(...) {}

  bool hasBeenUpdated();
}

class _MirrorBuildScriptUpdates implements BuildScriptUpdates {
  ...
}

 class _NoopBuildScriptUpdates implements BuildScriptUpates {
  @override
  bool hasBeenUpdated(_) => false;
}
```

Reply:
Done